sdks/python/gen_protos.py

```diff
@@ -123,7 +123,19 @@ def generate_urn_files(out_dir, api_path):
   This is executed at build time rather than dynamically on import to ensure
   that it is compatible with static type checkers like mypy.
   """
-  from google._upb import _message
+  try:
+    from google._upb import _message
+    list_types = (
+        list,
+        _message.RepeatedScalarContainer,
+        _message.RepeatedCompositeContainer,
+    )  # pylint: disable=c-extension-no-member
+  except ImportError:
+    from google.protobuf.internal import containers
```

Comment:
could you add a comment in which cases you expect else branch be used?

Reply:
Yes, that would be helpful.  I haven't caught this error while i was using when `PROTOCOL_BUFFERS_PYTHON_IMPLEMENTATION=upb`.
 but when `PROTOCOL_BUFFERS_PYTHON_IMPLEMENTATION==python`(slow implementation of protobufs which we don't want to use)  I was using  `except` case.

Reply:
I've refactored this to be more explicit. Apparently the default implementation depends on the platform (and what wheels are shipped for it). Fast protos shouldn't matter here, and we are (as we should) verifying that we have a compiled implementation in our containers.